Runtime/Components/WorldGen.cs

```diff
@@ -126,65 +127,68 @@ public void ClearChildren()
 
 		//Get 'True Raduis' or Diameter
         TRadius = Radius * 2 + 1;
-
 		DecorChanceMap = null;
-
 	}
 
-
-
-	//TODO Document This Function
     public void BuildWorld()
     {
-		ClearWorldData();
-		
+        ClearWorldData();
 
-		WorldGenStart.Invoke();
-			
-		//If we are not using a preset Seed generate a new one
-        if (!UseSeed)
-            Seed = Random.Range(int.MinValue,int.MaxValue);
-		Seeds = new int[OctaveCount + 2];
-		Seeds[0] = Seed;
-			
-		System.Random r = new System.Random(Seed);  //Create the Random object to be used
-		Seeds[OctaveCount + 1] = r.Next();
+        Init_Seeds();
+		WorldGenPreInit.Invoke();
 
-		//Intialize the Octaves
-		if (Octaves == null || Octaves.Length != OctaveCount)
-			Octaves = new TerrainNoiseObject[OctaveCount];
+        WorldGenStart.Invoke();
+        //Build the Tile Grid
+        Init_Tiles();
+    }
 
-		//Define the ScaleMap
-		PrepareNoise(ref ScaleMap);
-		ScaleMap.Intialize(Seed, 3);
-		//Construct the Octaves
-        for (int OctaveIndex = 0; OctaveIndex < OctaveCount; OctaveIndex++)
-        {
-            Seeds[OctaveIndex+1] = r.Next();//Get the Seed for Each Octave
-			PrepareNoise(ref Octaves[OctaveIndex]);
-            Octaves[OctaveIndex].Intialize(Seeds[OctaveIndex], Mathf.RoundToInt(Mathf.Pow(5, OctaveIndex)));//New Terrain Noise (Seed, Grid Size(5^Octave)
-        }
-		
-		//Build the Tile Grid
-		Tiles = new GameObject[TRadius * TRadius];  //Define The tiles in code
 
+    private void Init_Tiles()
+    {
+        Tiles = new GameObject[TRadius * TRadius];  //Define The tiles in code
 
-		//Actually Build the Tiles
-		for (int y = Radius * -1; y <= Radius; y++)
+        //Actually Build the Tiles
+        for (int y = Radius * -1; y <= Radius; y++)
         {
             for (int x = Radius * -1; x <= Radius; x++)
             {
-				int id = (Radius - y) * TRadius + (x + Radius);
+                int id = (Radius - y) * TRadius + (x + Radius);
```

Comment:
So looking at this line I see an imediate problem, need to rename either Radius or TRadius, What is TRadius and how does it differ?

Reply:
This will be resolved next sprint